src/Doctrine/Common/State/LinksHandlerTrait.php

```diff
@@ -41,7 +43,8 @@ private function getLinks(string $resourceClass, Operation $operation, array $co
         $linkProperty = $context['linkProperty'] ?? null;
 
         foreach ($links as $link) {
-            if ($linkClass === $link->getFromClass() && $linkProperty === $link->getFromProperty()) {
+            $testedProperty = $this->nameConverter ? $this->nameConverter->normalize($link->getFromProperty()) : $link->getFromProperty();
+            if ($linkClass === $link->getFromClass() && $linkProperty === $testedProperty) {
```

Comment:
not sure I understand, what's your resource configuration? why isn't the `fromProperty` already the correct name? this should happend way before reaching this code, for example in a MetadataFactory, although I don't get the use case, just configure the property manually?

Reply:
My resource configuration is minimalist, `#[ApiResource]`. The `fromProperty` is set during the `cache:clear`, using the default `LinkFactory` that reads the relations. If the link `fromProperty` is set to the serialized form, the mapping can not be recovered in the `getAssociationMapping` call (https://github.com/api-platform/core/blob/main/src/Doctrine/Orm/State/LinksHandlerTrait.php#L80).

Reply:
Is it possible to put the name converter inside https://github.com/api-platform/core/blob/main/src/Metadata/Resource/Factory/LinkResourceMetadataCollectionFactory.php#L44 instead ? (or in the link factory) like this it'd be cached

Reply:
The LinkFactory must not set the `fromProperty` to the normalized form because it would break the getAssociationMapping call (https://github.com/api-platform/core/blob/main/src/Doctrine/Orm/State/LinksHandlerTrait.php#L80).

The only alternative I see (if we want to use the cache) would be to add another property to the `Link`, having both `fromProperty` and (e.g.) `fromPropertyNormalized`, both set in the Link Factory. But it looks like a quite heavy change.

Reply:
I'm having a hard time understanding the whole logic behind this there's something wrong... links should only handle property mapping, in the code nothing looks like `bar_converted` so this needs to be before we reach that part of the code. It's the request body itself that should be transformed to give `barConverted` at that point.

Reply:
OK. Yes I guess we can do the other way around, and use `serializer->denormalize(...)` when `$context['linkProperty']` is set (in https://github.com/api-platform/core/blob/3.1/src/GraphQl/Resolver/Stage/ReadStage.php#L86-L86 if I understand correctly)

Reply:
mhh probably can you give me a bit more time? I need to take a deeper look into that but I won't be able to until oct/nov...

Reply:
Yes, I am using API Platform in prod with this exact use case and I see no workaround so I will use my fork in the meantime